rustls/src/crypto/aws_lc_rs/hpke.rs

```diff
@@ -30,7 +32,7 @@ pub static DEFAULT_PROVIDER: &'static dyn HpkeProvider = &Provider {
 };
 
 /// All supported HPKE suites.
-pub static ALL_SUPPORTED_SUITES: &[&Suite] = &[
+pub static ALL_SUPPORTED_SUITES: &[&dyn AwsLcRsHpkeSuite] = &[
```

Comment:
Couldn't/shouldn't these have been `&dyn Hpke` from the start? And if so, do we still need the intermediate trait?

Reply:
It might be possible if we change the `Hpke` trait to offer a way to get a corresponding suite ID. I'm not 100% sure. In general I was trying to avoid changing elements that already landed. I think we'd also have to break semver for that change since there isn't a sensible default impl to use.